### PR TITLE
Auth DNSSEC: Make default options singular and use defaults in Cryptokey API-endpoint

### PR DESCRIPTION
**Pull request by pieterlexis**
### Short description
Before, the `default-{k,z}sk-algorithms` options would accept multiple options. While the `default-{k,z}sk-size` would only accept a single value.

The first commit makes the algorithms options singular.

Furthermore, size is only needed for non-fixed length algo's (and we don't have defaults). the second displays a warning on start up if the size is 0 when it should be a number.

The last commit makes the cryptokey endpoint in the API use the defaults if `algo` and/or `size` are not passed in the request.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled and tested this code
- [x] included documentation (including possible behaviour changes)
- [x] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: docs/http-api/endpoint-cryptokeys.rst [hunk 1]
```diff
@@ -17,6 +17,11 @@ These endpoints allow for the manipulation of DNSSEC crypto material.
   This method adds a new key to a zone.
   The key can either be generated or imported by supplying the ``content`` parameter.
 
+  if ``content``, ``bits`` and ``algo`` are null, a key will be generated based
+  on the :ref:`setting-default-ksk-algorithm` and :ref:`setting-default-ksk-size`
+  sttings for a KSK and the :ref:`setting-default-zsk-algorithm` and :ref:`setting-default-zsk-size`
```
Comment (rgacogne) [inline docs/http-api/endpoint-cryptokeys.rst]: sttings -> settings

File: docs/settings.rst [hunk 2]
```diff
@@ -272,13 +272,17 @@ Debugging switch - don't use.
...
 
-``default-ksk-algorithms``
+``default-ksk-algorithm``
 --------------------------
 
 -  String
 -  Default: ecdsa256
 
+.. versionchanged:: 4.1.0
+  Renamed from ``default-ksk-algorithms``. Does no longer support multiple algorithm names.
```
Comment (Habbie) [inline docs/settings.rst]: No longer supports multiple ..

File: docs/settings.rst [hunk 3]
```diff
@@ -272,13 +272,17 @@ Debugging switch - don't use.
...
+
 The algorithm that should be used for the KSK when running
 :doc:`pdnsutil secure-zone <manpages/pdnsutil.1>`. Must be one
```
Comment (Habbie) [inline docs/settings.rst]: or the API

File: docs/settings.rst [hunk 4]
```diff
@@ -301,9 +305,10 @@ of:
 --------------------
 
 -  Integer
--  Default: whichever is default for ``default-ksk-algorithms``
+-  Default: whichever is default for `default-ksk-algorithm`_
 
 The default keysize for the KSK generated with :doc:`pdnsutil secure-zone <dnssec/pdnsutil>`.
+Only relevant for algorithms with non-fixed keysizes (like RSA)
```
Comment (Habbie) [inline docs/settings.rst]: missing period at end

File: pdns/common_startup.cc [hunk 5]
```diff
@@ -524,6 +525,17 @@ void mainthread()
...
 
+  // Some sanity checking on default key settings
+  for (const string& algotype : {"ksk", "zsk"}) {
+    int algo, size;
+    algo = DNSSECKeeper::shorthand2algorithm(::arg()["default-"+algotype+"-algorithm"]);
+    size = ::arg().asNum("default-"+algotype+"-size");
+    if (algo == -1)
+      L<<Logger::Warning<<"Warning: default-"<<algotype<<"-algorithm set to unknown algorithm: "<<::arg()["default-"+algotype+"-algorithm"]<<endl;
+    else if (algo <= 10 && size == 0)
+      L<<Logger::Warning<<"Warning: default-"<<algotype<<"-algorithm is set to an algorithm("<<::arg()["default-"+algotype+"-algorithm"]<<") that requires a non-zero default-"<<algotype<<"-size!"<<endl;
```
Comment (Habbie) [inline pdns/common_startup.cc]: s/algorithm(/algorithm (/

File: pdns/common_startup.cc [hunk 6]
```diff
@@ -183,9 +184,9 @@ void declareArguments()
 
   ::arg().setSwitch("traceback-handler","Enable the traceback handler (Linux only)")="yes";
   ::arg().setSwitch("direct-dnskey","Fetch DNSKEY RRs from backend during DNSKEY synthesis")="no";
-  ::arg().set("default-ksk-algorithms","Default KSK algorithms")="ecdsa256";
+  ::arg().set("default-ksk-algorithm","Default KSK algorithms")="ecdsa256";
```
Comment (mind04) [inline pdns/common_startup.cc]: s/algorithms/algorithm in the description 4x (here and below)

**Comment by zeha** [comment]
API bits LGTM

**Pull request merged**